pyccel/stdlib/cwrapper/cwrapper.c

```diff
@@ -0,0 +1,400 @@
+	return true;
+}
+
+/*
+ * Function: _check_pyarray_type
+ * --------------------
+ * Check if Python Object is ArrayType:
+ *
+ * 	Parameters	:
+ *		a 	  : python array object
+ *
+ * 	Returns		:
+ *		return true if no error occurred otherwise it will return false
+ *      and raise TypeError exception
+ * reference of the used c/python api function
+ * -------------------------------------------
+ * https://numpy.org/doc/stable/reference/c-api/array.html#c.PyArray_Check
+ */
+
+static bool _check_pyarray_type(PyObject *a)
+{
+	if (!PyArray_Check(a))
+	{
+		PyErr_Format(PyExc_TypeError,
+			"argument must be numpy.ndarray, not %s",
+			 a == Py_None ? "None" : Py_TYPE(a)->tp_name);
+		return false;
+	}
+
+	return true;
+}
+
+/*
+ * Function: _check_array
+ * --------------------
+ * Check Python Object (DataType, Rank, Order):
+ *
+ * 	Parameters	:
+ *		a 	  : python array object
+ *      dtype : desired data type enum
+ *		rank  : desired rank
+ *		flag  : desired order flag
+ * 	Returns		:
+ *		return true if no error occurred otherwise it will return false
+ */
+
+static bool	_check_array(PyArrayObject *a, int dtype, int rank, int flag)
+{
+	if(!check_pyarray_dtype(a, dtype)) return false;
+
+	if(!_check_pyarray_rank(a, rank)) return false;
+
+	if(rank > 1 && !_check_pyarray_order(a, flag)) return false;
+
+	return true;
+}
+
+/*
+ * Function: pyarray_to_ndarray
+ * ----------------------------
+ * A Cast function that convert numpy array variable into ndarray variable,
+ * by copying its information and data to a new variable of type ndarray struct
+ * and return this variable to be used inside c code.
+ * 	Parameters	:
+ *		o 	  : python array object
+ *      array : c ndarray object
+ *      dtype : desired data type enum
+ *		rank  : desired rank
+ *		flag  : desired order flag
+ * 	Returns		:
+ *		return true if no error occurred otherwise it will return false
+ *
+ * reference of the used c/numpy api function
+ * -------------------------------------------
+ * https://numpy.org/doc/stable/reference/c-api/array.html
+ */
+
+bool	pyarray_to_ndarray(PyObject *o, t_ndarray *array, int dtype, int rank, int flag)
+{
+	PyArrayObject	*pyarray;
+
+	// Importing the API
+	if (PyArray_API == NULL) import_array();
+	// Array type c
+	if (!_check_pyarray_type(o)) return false;
+
+	pyarray = (PyArrayObject *)o;
+
+	if (!_check_array(pyarray, dtype, rank, flag)) return false;
+
+	array->nd          = PyArray_NDIM(pyarray);
+	array->raw_data    = PyArray_DATA(pyarray);
+	array->type_size   = PyArray_ITEMSIZE(pyarray);
+	array->type        = PyArray_TYPE(pyarray);
+	array->length      = PyArray_SIZE(pyarray);
+	array->buffer_size = PyArray_NBYTES(pyarray);
+
+	array->shape       = (int64_t*)malloc(sizeof(int64_t) * array->nd);
+	memcpy(array->shape, PyArray_SHAPE(pyarray), sizeof(int64_t) * array->nd);
+
+	array->strides     = (int64_t*)malloc(sizeof(int64_t) * array->nd);
+	memcpy(array->strides, PyArray_STRIDES(pyarray), sizeof(int64_t) * array->nd);
```

Comment:
The shape and stride cannot be handled with memcpy. See [ndarrays.c](https://github.com/pyccel/pyccel/blob/master/pyccel/stdlib/ndarrays/ndarrays.c#L326-L351) and issue #758

Reply:
ah, I forget about it thx